src/controllers/dynakube/activegate/reconciler.go

```diff
@@ -0,0 +1,139 @@
+package activegate
+
+import (
+	"context"
+	"fmt"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	AGSecretSuffix             = "-activegate-tenant-secret"
+	CommunicationEndpointsName = "communication-endpoints"
+	TenantTokenName            = "tenant-token"
+	TenantUuidName             = "tenant-uuid"
+)
+
+type Reconciler struct {
```

Comment:
The name of `Reconciler` is to general here, this only makes sure that the TenantSecret is present and up to date.

`TenantSecretReconciler` or something would make it clearer.

Also putting `AG` into all function names feels a bit unnecessary